src/lib/fauna-client.ts

```diff
@@ -57,57 +44,39 @@ export default class FaunaClient {
       typecheck: opts?.typecheck ?? undefined,
       secret: opts?.secret ?? this.secret,
     };
-    return new Promise((resolvePromise, rejectPromise) => {
-      let req: any;
-      const onResponse = (http2ResponseHeaders: any) => {
-        const status = http2ResponseHeaders[constants.HTTP2_HEADER_STATUS];
-        let responseData = "";
+    const url = new URL(this.endpoint);
+    url.pathname = "/query/1";
+    const res = await fetch(url, {
+      method: "POST",
+      headers: {
+        authorization: `Bearer ${secret ?? this.secret}`,
+        "x-fauna-source": "Fauna Shell",
+        ...(typecheck !== undefined && { "x-typecheck": typecheck.toString() }),
+        ...(format !== undefined && { "x-format": format }),
+      },
+      body: JSON.stringify({ query }),
+    });
 
-        req.on("data", (chunk: any) => {
-          responseData += chunk;
-        });
+    const json = await res.json();
 
-        req.on("end", () => {
-          resolvePromise({
-            status,
-            body: JSON.parse(responseData),
-            headers: http2ResponseHeaders,
-          });
-        });
+    if (res.status === 200 || res.status === 201) {
+      return {
+        status: 200,
+        body: {
+          data: json.data as T,
+        },
```

Comment:
this should just be
```suggestion
        body: json,
```

so that the `summary` field gets copied over. for example, try this: `fauna eval "log('hi')"`

Reply:
o I see, thats why the summary field was shared across error and success types

Reply:
yeah. you can also get log messages back in the `summary` for runtime errors too:
```ts
log("hi"); 1/0

// returns

divide_by_zero: Attempted integer division by zero.

error: Attempted integer division by zero.
at *query*:1:14
  |
1 | log('hi'); 1/0
  |              ^
  |

info at *query*:1: hi
```